File: src/GLFMmatlab/mapping_functions/f_g_1.m
function y = f_g_1(x, mu, w)
    % transformation function for real-valued data
    % X -> Y (from data to pseudo-obversations)
    if (w == 0)
        error('scaling factor should never be 0');
    end
    y = w .* (x - mu);
end

File: src/GLFMR/f_g1.R

#' @description  Transformation function for real-valued data
#' Y -> X (from pseudo-obversations to data)
#' @param y: N*R where N is the number of observations and R is the num of categories
#' @param mu: mean to standarise data
#' @param w: weights
#' @Return x standarised data

f_g1<function(x, mu, w){
  if (w == 0){
    stop('scaling factor should never be 0')
  }
  else{
    y <- w * (x - mu)
  }
}

